[core] limit metric cardinality aggregation for tasks and actors metrics

can-anyscale opened the pull request:
Currently, for metric cardinality aggregation, after dropping certain labels (such as WorkerId), we aggregate samples by summing their values. However, this strategy is not correct universally, for example, latency gauge metrics should not be summed across workers, but rather sampled by average, mean, etc.

To avoid incorrect aggregation, limit the current cardinality-reduction logic to only tasks and actors metrics, where summing is the appropriate aggregation method.

Test:
- CI

Commit:
[core] limit metric cardinality aggregation for tasks and actors metrics

Signed-off-by: Cuong Nguyen <can@anyscale.com>

jjyao reviewed (approved):
Could you create a ticket for the proper fix?

can-anyscale committed:
[core] aggregation function for high cardinality metric (#59179)

`RAY_metric_cardinality_level` is a setting a user can set to reduce
high cardinality label in Ray metrics. By default, `WorkerId` label will
be dropped and the values will be aggregated using a sum function. This
only supports `ray_tasks` and `ray_actors` metric.

This PR supports configurable aggregation function. For example, for a
gauge latency metric, one might want to use `median` etc. rather than
`sum` as an aggregation.

Test:
- CI

---------

Signed-off-by: Cuong Nguyen <can@anyscale.com>

can-anyscale committed:
Merge branch 'master' into can-car01
